hrms/hr/doctype/shift_assignment/shift_assignment.py

```diff
@@ -35,6 +40,29 @@ def validate_overlapping_shifts(self):
 			if overlapping_timings:
 				self.throw_overlap_error(overlapping_dates[0])
 
+	def validate_same_date_multiple_shifts(self):
+		overlapping_dates = self.get_overlapping_dates()
+		if not len(overlapping_dates):
+			return
+
+		allow_multiple_shift_assignments = frappe.db.get_single_value(
+			"HR Settings", "allow_multiple_shift_assignments"
+		)
+		if not allow_multiple_shift_assignments:
+			frappe.throw(
+				msg=_(
+					"Multiple shift assignments for the same date has been disabled. Please enable this feature under {0}."
```

Comment:
```suggestion
					"{employee} already has an active Shift Assignment {ID} for some/all of these dates. To allow this enable 'Allow Multiple Shift Assignments' for the same date in {HR Settings link}."
```
Let's show at least one conflicting assignment's ID in the message so that the user doesn't have to search for it.
Also, add title to the message